XMonad/Hooks/ScreenCorners.hs

```diff
@@ -51,31 +55,46 @@ data ScreenCorner
 -- ExtensibleState modifications
 --------------------------------------------------------------------------------
 
-newtype ScreenCornerState = ScreenCornerState (M.Map Window (ScreenCorner, X ()))
+newtype ScreenCornerState = ScreenCornerState (M.Map (ScreenCorner, Int) (Window, X ()))
 
 instance ExtensionClass ScreenCornerState where
   initialValue = ScreenCornerState M.empty
 
 -- | Add one single @X ()@ action to a screen corner
+{-# DEPRECATED addScreenCorner "addScreenCorner works only in a single monitor setup. Use addMonitorCorner instead." #-}
 addScreenCorner :: ScreenCorner -> X () -> X ()
 addScreenCorner corner xF = do
   ScreenCornerState m <- XS.get
-  (win, xFunc) <- case find (\(_, (sc, _)) -> sc == corner) (M.toList m) of
-    Just (w, (_, xF')) -> return (w, xF' >> xF) -- chain X actions
+  let key = (corner, -1) -- Use -1 to indicate a non-monitor-specific corner
+  (win, xFunc) <- case M.lookup key m of
+    Just (w, xF') -> return (w, xF' >> xF) -- Chain X actions
     Nothing -> (,xF) <$> createWindowAt corner
-
-  XS.modify $ \(ScreenCornerState m') -> ScreenCornerState $ M.insert win (corner, xFunc) m'
+  XS.modify $ \(ScreenCornerState m') -> ScreenCornerState $ M.insert key (win, xFunc) m'
 
 -- | Add a list of @(ScreenCorner, X ())@ tuples
+{-# DEPRECATED addScreenCorners "addScreenCorners works only in a single monitor setup. Use addMonitorCorners instead." #-}
 addScreenCorners :: [(ScreenCorner, X ())] -> X ()
 addScreenCorners = mapM_ (uncurry addScreenCorner)
 
+-- | Add one single @X ()@ action to a screen corner on a specific monitor
+addMonitorCorner :: ScreenCorner -> Int -> Dimension -> X () -> X ()
+addMonitorCorner corner monitorNumber hotZoneSize xF = do
```

Comment:
For consistency's sake, I think a version of this with a default size of 1 might be nice